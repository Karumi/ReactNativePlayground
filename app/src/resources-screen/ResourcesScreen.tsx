import { Container } from "native-base";
import * as React from "react";
import { Image } from "react-native";
import Toolbar from "../base-components/toolbar/Toolbar";
import translator from "../i18n";
import assets from "./images/assets";

export interface Props {
    navigation: any;
}
class ResourcesScreen extends React.Component<Props> {
    public render() {
        return (
            <Container>
                <Toolbar
                    title={translator().resourcesScreenTitle}
                    navigation={this.props.navigation}
                    showBackButton={true}
                />
                <Image source={assets.anyAsset} style={{ height: 200, width: 200 }} />
                <Image source={assets.arrow} style={{ height: 20, width: 20}} />
            </Container>
        );
    }
}

export default ResourcesScreen;
