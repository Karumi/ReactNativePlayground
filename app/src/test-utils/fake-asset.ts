jest.mock("../assets/asset", () => {
return {
        default: (uri) => uri,
    };
});
