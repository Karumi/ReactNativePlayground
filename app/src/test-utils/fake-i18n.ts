
import { I18nResolver } from "i18n-ts";
import en from "../translations/en";

const i18n = { default: en };

jest.mock("../i18n", () => {
return {
        default: () => new I18nResolver(i18n,  "en").translation,
    };
} );
