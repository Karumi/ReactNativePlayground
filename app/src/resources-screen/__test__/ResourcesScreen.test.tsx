import * as React from "react";
import * as renderer from "react-test-renderer";
import "../../test-utils/fake-i18n";
import ResourcesScreen from "../ResourcesScreen";

jest.mock("../images/assets", () => {
    return {
        default: {
            arrow: "arrow",
        },
    };
});

jest.mock("../../base-components/toolbar/Toolbar", () => {
    return {
        default: "Toolbar",
    };
});

it("renders correctly with defaults", () => {
    const toolbar = renderer.create(<ResourcesScreen navigation={jest.fn()}/>).toJSON();
    expect(toolbar).toMatchSnapshot();
});
