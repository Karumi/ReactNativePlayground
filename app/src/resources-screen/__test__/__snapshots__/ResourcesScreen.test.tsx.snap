// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <Toolbar
    navigation={[MockFunction]}
    showBackButton={true}
    title="Resources"
  />
  <Image
    source="any-asset"
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
  <Image
    source="arrow"
    style={
      Object {
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;
